src/pages/StockDetailPage.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCompany, clearCompany } from '../redux/company/company';
+import LineChart from '../components/LineChart';
+import BarChart from '../components/BarChart';
+import { grossData, netIncomeData } from '../utils/utils';
+import ErrorPage from './ErrorPage';
+import Spinner from '../components/Spinner';
+import '../style/stockDetailsPage.css';
+
+const StockDetailPage = () => {
+  const params = useParams();
+  const dispatch = useDispatch();
+  const { symbol } = params;
+  useEffect(() => {
+    dispatch(clearCompany());
+    dispatch(fetchCompany(symbol));
+  }, []);
+
+  const state = useSelector((state) => state.companyReducer);
+  const grossResult = grossData(state.data);
+  const netResult = netIncomeData(state.data);
+  const { loading, error } = state;
+
+  if (error) {
+    return (
+      <>
+        <ErrorPage />
+      </>
+    );
+  }
+
+  if (loading) {
+    return (
+      <>
+        <Spinner />
+      </>
+    );
+  }
```

Comment:
- I see you've used this logic across multiple components to decide whether the component will display the load page or the error page before rendering. I have a suggestion to save it in a separate function in some file `./loadCondition' and then reuse it whenever you want. This makes less and cleaner code!